### PR TITLE
Correctly set the Hints for zimrecreate.

### PR DESCRIPTION
**Pull request by mgautierfr**
Needed for (and need) openzim/libzim#466

### CODE REVIEW

File: src/tools.h [hunk 1]
```diff
@@ -156,6 +156,10 @@ class CopyItem : public zim::writer::Item         //Article class that will be p
     {
        return std::unique_ptr<zim::writer::ContentProvider>(new ItemProvider(item));
     }
+
+    zim::writer::Hints getHints() const {
+      return {{ zim::writer::HintKeys::FRONT_ARTICLE, (item.getMimetype() == "text/html")}};
+    }
```
Comment (kelson42) [inline src/tools.h:162]: @mgautierfr What about if the mimetype has additional attributes like `charset=utf8`? This won't be detected properly?

Comment (mgautierfr) [reply src/tools.h]: I've missed that.
But in the same time, we probably don't want to put `text/html;raw=true` as front article.

Last push fix this.

---

On a more global way, as commented [here](https://github.com/kiwix/kiwix-lib/pull/403#issuecomment-671277426), we face the problem that we have to parse something we shouldn't. (either for `raw=true` than `text/html`)

Hopefully, the (next to implement) extra data feature will fix this.

File: src/tools.h [hunk 2]
```diff
@@ -156,6 +156,13 @@ class CopyItem : public zim::writer::Item         //Article class that will be p
     {
        return std::unique_ptr<zim::writer::ContentProvider>(new ItemProvider(item));
     }
+
+    zim::writer::Hints getHints() const {
+      auto mimetype = item.getMimetype();
+      auto front_article = ( mimetype.find("text/html") != std::string::npos
+                          && mimetype.find("raw=true") == std::string::npos);
```
Comment (veloman-yunkan) [inline src/tools.h:163]: Convert this piece of code to a free helper function `bool is_front_article(const zim::Item& item)` and reuse it in `PatchItem`.

**Review by veloman-yunkan — CHANGES_REQUESTED** [review]

File: src/tools.h (hunk 2 above)
Comment (mgautierfr) [reply src/tools.h]: Done. Using name `guess_is_front_article` because it is a guess. mimetype is not a real criteria, but we don't have best.

**Review by veloman-yunkan — APPROVED** [review]

**Comment by kelson42** [comment]
@mgautierfr I have rebased, but testing seems insufficient

**Comment by mgautierfr** [comment]
I've updated a bit the code :
- The helper function now takes a "mimetype" string instead of a item. It allows us to unit test it.
- The mimetype must start by "text/html", not just containing it.
- I have added unit test on method `guess_is_front_article`

The coverage is pretty low as we don't run zim-recreate at all in the unit tests.
And all the "writerItem" methods are called by `zim::writer::Creator` when we actually recreate a zim.

**Review by veloman-yunkan — APPROVED** [review]

**Pull request merged**